Fix building LuaJIT on Linux

kromych opened the pull request:
**Describe the pull request**

- #### What does your PR fix?
 Fixes building LuaJIT on Linux, issue [24502](https://github.com/microsoft/vcpkg/issues/24502)

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  * `x64-linux`, `x64-windows`, `x64-windows-static`, `x64-windows-static-md`, `x86-windows` are supported
  * `Yes, I have`

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  `Yes, it does`

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  `Yes, I have`

kromych committed:
Fix building LuaJIT on Linux

kromych committed:
Update version metadata

BillyONeal commented on ports/luajit/portfile.cmake:
```diff
@@ -20,21 +20,38 @@ if (NOT VCPKG_BUILD_TYPE OR VCPKG_BUILD_TYPE STREQUAL debug)
+        endif()
+        vcpkg_copy_pdbs()
+    else()
+        vcpkg_execute_build_process(
```
Should this use `vcpkg_build_make`?

dg0yt replied:
`vcpkg_build_make` is tightly coupled to `vcpkg_configure_make`, not a pure function.

BillyONeal committed:
Fix remaining ci.baseline.txt entries:

luajit:arm64-windows: Left as "skip": The problem is that the batch files used to build here want to run a binary they built which fails, but there's no good way to say "I can run the results" in our platform expressions right now
luajit:arm64-uwp / luajit:arm-uwp: These are already stopped by the platform expression.
luajit:x64-osx: I don't know if this works yet but we should try.

BillyONeal committed:
Block osx with a "supports".
